deepchem/data/datasets.py

```diff
@@ -156,43 +248,83 @@ def get_task_names(self):
 
   @property
   def X(self):
-    """Get the X vector for this dataset as a single numpy array."""
+    """Get the X vector for this dataset as a single numpy array.
```

Comment:
On this and the following properties, how about adding a note that accessing it may involve loading data from disk, and that using `iterbatches()` or `itersamples()` can be more efficient.

Reply:
Good point! Added in a note